iceoryx_posh/include/iceoryx_posh/mepoo/chunk_header.hpp

```diff
@@ -37,7 +37,7 @@ struct alignas(32) ChunkHeader
     void* payload() const
     {
         // payload is always located relative to "this" in this way
-        return reinterpret_cast<void*>(reinterpret_cast<uintptr_t>(this) + sizeof(ChunkHeader));
+        return reinterpret_cast<void*>(reinterpret_cast<uint64_t>(this) + sizeof(ChunkHeader));
```

Comment:
What are the consequences for 32-bit architectures?

Reply:
We are unable to use the uintptr_t since it is a typedef to unsigned long, unsigned long long or just unsigned - every platform and every libc implementations seems to handle it differently. If you take a look for instance here: https://en.cppreference.com/w/cpp/types/integer you see that an uintptr_t is an "unsigned integer type capable of holding a pointer" which is exactly what we require here.

An uint64_t is therefore the perfect choice since it is able to hold a pointer for 32-bit and 64-bit architectures.